### PR TITLE
Feature/stream 1000

### PR DESCRIPTION
**Pull request by will-placester**
Created Delete stream, Create stream. Will add more streams.

### CODE REVIEW

File: main.tf [hunk 1]
```diff
@@ -1,13 +1,29 @@
-variable "DeliveryStreamName" {
-  description = "Name of the Firehose delivery stream"
+provider "aws" {
+  region = "${var.aws_region}"
```
Comment (mrapptambour) [inline main.tf]: @will-placester is aws_region a called out input variable

File: main.tf [hunk 2]
```diff
@@ -1,13 +1,29 @@
...
+    Prefix             = "${var.Prefix}"
+    LogStreamName      = "${var.LogStreamName}"
+    LogGroupName       = "${var.LogGroupName}"
+    BackupPrefix       = "${var.BackupPrefix}"
+    BucketARN          = "${aws_s3_bucket.property_canonical_bucket.arn}"
+    BackupBucketARN    = "${aws_s3_bucket.property_extracted_bucket.arn}"
+    LambdaArn          = "${aws_lambda_function.transformer.arn}"
+    LambdaArnName      = "${aws_lambda_function.transformer.function_name}"
+
+    BufferingHints = {
```
Comment (mrapptambour) [inline main.tf]: @will-placester have you confirmed that passing in a map object like this to CF works? I thought it was only KV params.

File: src/index.js [hunk 3]
```diff
@@ -6,36 +6,36 @@ exports.handler = function (event, context) {
...
-          Enabled: true || false,
+          Enabled: true,
           Processors: [
             {
               Type: 'Lambda', /* required */
               Parameters: [
                 {
-                  ParameterName: 'LambdaArn | NumberOfRetries', /* required */
-                  ParameterValue: 'STRING_VALUE' /* required */
+                  ParameterName: event.ResourceProperties.LambdaArnName, /* required */
```
Comment (mrapptambour) [inline src/index.js]: Looks like you could just hard code this to LambdaArn name. And do something like if (event.ResourceProperties.NumberOfRetries) then append 'NumberOfRetries': value to Parameters

File: src/index.js [hunk 4]
```diff
@@ -44,27 +44,27 @@ exports.handler = function (event, context) {
...
           },
           CloudWatchLoggingOptions: {
             Enabled: true || false,
-            LogGroupName: 'STRING_VALUE',
-            LogStreamName: 'STRING_VALUE'
+            LogGroupName: event.ResourceProperties.LogGroupName,
+            LogStreamName: event.ResourceProperties.LogStreamName
           },
-          CompressionFormat: 'UNCOMPRESSED | GZIP | ZIP | Snappy',
+          CompressionFormat: 'UNCOMPRESSED',
```
Comment (mrapptambour) [inline src/index.js]: @will-placester pass in from TF params, TF can default it to UNCOMPRESSED

File: src/index.js [hunk 5]
```diff
@@ -44,27 +44,27 @@ exports.handler = function (event, context) {
...
+            // KMSEncryptionConfig: {
+            //   AWSKMSKeyARN: 'STRING_VALUE' /* required */
+            // },
             NoEncryptionConfig: 'NoEncryption'
           },
-          Prefix: 'STRING_VALUE'
+          Prefix: event.ResourceProperties.Prefix
         },
-        S3BackupMode: 'Disabled | Enabled'
+        S3BackupMode: 'Enabled'
```
Comment (mrapptambour) [inline src/index.js]: @will-placester same, can probably have it passed in from TF with a default to Enabled

File: stack.yml [hunk 6]
```diff
@@ -5,25 +5,39 @@ Parameters:
...
       Handler: index.handler
-      Role:
-      Description: Custom CFN resource to creates and deletes an AWS Firehose Delivery Steam with an ExtendedS3DestinationConfiguration
+      Role: Ref RoleARN
+      Description: Custom CFN resource to creates and deletes an AWS Firehose Delivery Stream with an ExtendedS3DestinationConfiguration
       Runtime: nodejs4.3
       Timeout: 300
       Code:
-        ZipFile: !Ref LambdaFunctionCode
+        ZipFile: !Ref LambdaFunctionName
```
Comment (mrapptambour) [inline stack.yml]: This still be LambdaFunctionCode no? It's the raw code from the index.js file.

File: main.tf [hunk 7]
```diff
@@ -1,13 +1,29 @@
...
+      SizeInMBs         = "${var.SizeInMBs}"
+      IntervalInSeconds = "${var.IntervalInSeconds}"
+    }
+
+    FunctionCode = "${file("${path.module}/src/index.js")}"
   }
 
   template_body = "${file("${path.module}/stack.yaml")}"
+  policy_body   = "${file("${path.module}/templates/firehose_delivery_role.json")}"
```
Comment (mrapptambour) [inline main.tf]: @will-placester this needs to migrate to data.template_file.firehose_policy.rendered. You use the `template_file` resource to pass params to a templated file, then TF fulfills the body of it.

File: stack.yml (hunk 6 above)
Comment (will-placester) [inline stack.yml]: Seems I forgot to commit that change!

File: main.tf (hunk 1 above)
Comment (will-placester) [inline main.tf]: NOPE, can be removed

File: main.tf (hunk 7 above)
Comment (will-placester) [inline main.tf]: I can use URL it appears instead, should work....

**Pull request merged**